### PR TITLE
Can't delete class based apps

### PR DESCRIPTION
**Pull request by Muniraj132**
@incraigulous  I have created the pull request for "Can't delete class based apps" task. 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208138402817782

### CODE REVIEW

File: src/Controllers/Admin/AppSettingsController.php [hunk 1]
```diff
@@ -413,4 +463,62 @@ public function delete_app( $slug )
 
         return $response;
     }
+
+    /** Soft delete an app by its slug.
+     * This function marks an app as soft deleted based on its slug.
+     *  @param string $slug The slug of the app to be soft deleted.
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse Redirects to the admin page with a success message.
+     */
+     public function softDelete_app( $slug)
```
Comment (incraigulous) [inline src/Controllers/Admin/AppSettingsController.php:472]: @Muniraj132 , small code style issue here, this should be either softdelete_app, or soft_delete_app according to WordPress code style.

Comment (incraigulous) [reply src/Controllers/Admin/AppSettingsController.php:472]: Also, please add a space between $slug and )

**Review by incraigulous — CHANGES_REQUESTED** [review]
Thanks @Muniraj132, small code style change requested!

File: src/Controllers/Admin/AppSettingsController.php (hunk 1 above)
Comment (Muniraj132) [reply src/Controllers/Admin/AppSettingsController.php:484]: @incraigulous Thank you for pointing that out. I'll update the code to follow the WordPress coding standards, changing it to either softdelete_app or soft_delete_app as suggested. I'll also add the space between $slug and ).

**Review by incraigulous — APPROVED** [review]

**Pull request merged**